SDL build is failing when ENABLE_LOG=OFF

### Bug Report
SDL build is failing when ENABLE_LOG=OFF

##### Reproduction Steps
1. Check out commit af97446e7e213997174f7fa10fabf8b86f316562 from current develop
2. Build with `cmake .. -DENABLE_LOG=OFF,` and `make`
3. Build has failed

##### Expected Behavior
SDL build should be OK with ENABLE_LOG=OFF

##### Observed Behavior
SDL build has failed
```
16:29:51 Scanning dependencies of target HMI_API
16:29:51 [  5%] Building CXX object src/components/interfaces/CMakeFiles/HMI_API.dir/HMI_API_schema.cc.o
16:30:07 In file included from /home/jenkins/workspace/develop_weekly_multibuild/ENABLE_LOG/OFF/POLICY/HTTP/label/build/build/src/components/interfaces/HMI_API_schema.cc:49:0:
16:30:07 /home/jenkins/workspace/develop_weekly_multibuild/ENABLE_LOG/OFF/POLICY/HTTP/label/build/src/components/smart_objects/include/smart_objects/enum_schema_item.h: In member function ���NsSmartDeviceLink::NsSmartObjects::Errors::eType NsSmartDeviceLink::NsSmartObjects::TEnumSchemaItem<EnumType>::validate(const NsSmartDeviceLink::NsSmartObjects::SmartObject&, rpc::ValidationReport*)���:
16:30:07 /home/jenkins/workspace/develop_weekly_multibuild/ENABLE_LOG/OFF/POLICY/HTTP/label/build/src/components/smart_objects/include/smart_objects/enum_schema_item.h:242:12: error: no match for ���operator<<��� (operand types are ���std::stringstream {aka std::basic_stringstream<char>}��� and ���const char [21]���)
16:30:07      stream << "Invalid enum value: " << Object.asInt();
16:30:07             ^
16:30:07 /home/jenkins/workspace/develop_weekly_multibuild/ENABLE_LOG/OFF/POLICY/HTTP/label/build/src/components/smart_objects/include/smart_objects/enum_schema_item.h:242:12: note: candidate is:
16:30:07 In file included from /usr/include/c++/4.9/string:52:0,
16:30:07                  from /usr/include/c++/4.9/stdexcept:39,
16:30:07                  from /usr/include/c++/4.9/array:38,
16:30:07                  from /usr/include/c++/4.9/tuple:39,
16:30:07                  from /usr/include/c++/4.9/bits/stl_map.h:63,
16:30:07                  from /usr/include/c++/4.9/map:61,
16:30:07                  from /home/jenkins/workspace/develop_weekly_multibuild/ENABLE_LOG/OFF/POLICY/HTTP/label/build/build/src/components/interfaces/HMI_API_schema.cc:39:
16:30:07 /usr/include/c++/4.9/bits/basic_string.h:2771:5: note: template<class _CharT, class _Traits, class _Alloc> std::basic_ostream<_CharT, _Traits>& std::operator<<(std::basic_ostream<_CharT, _Traits>&, const std::basic_string<_CharT, _Traits, _Alloc>&)
16:30:07      operator<<(basic_ostream<_CharT, _Traits>& __os,
16:30:07      ^
16:30:07 /usr/include/c++/4.9/bits/basic_string.h:2771:5: note:   template argument deduction/substitution failed:
16:30:07 In file included from /home/jenkins/workspace/develop_weekly_multibuild/ENABLE_LOG/OFF/POLICY/HTTP/label/build/build/src/components/interfaces/HMI_API_schema.cc:49:0:
16:30:07 /home/jenkins/workspace/develop_weekly_multibuild/ENABLE_LOG/OFF/POLICY/HTTP/label/build/src/components/smart_objects/include/smart_objects/enum_schema_item.h:242:15: note:   ���std::stringstream {aka std::basic_stringstream<char>}��� is not derived from ���std::basic_ostream<_CharT, _Traits>���
16:30:07      stream << "Invalid enum value: " << Object.asInt();
```

##### OS & Version Information
* OS/Version: Ubuntu 14.04 (docker)
* SDL Core Version: Commit af97446e7e213997174f7fa10fabf8b86f316562 from `develop` branch

##### Test Case, Sample Code, and / or Example App
http://opensdl-jenkins.luxoft.com:8080/job/develop_weekly_multibuild/ENABLE_LOG=OFF,POLICY=HTTP,label=build/26/consoleFull

--- a/src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/put_file_request.cc
+++ b/src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/put_file_request.cc
@@ -212,10 +212,9 @@ void PutFileRequest::Run() {
     return;
   }
   const std::string full_path = file_path + "/" + sync_file_name_;
-  const size_t bin_data_size = binary_data.size();
 
   if ((*message_)[strings::msg_params].keyExists(strings::crc32_check_sum)) {
-    LOG4CXX_TRACE(logger_, "Binary Data Size:  " << bin_data_size);
+    LOG4CXX_TRACE(logger_, "Binary Data Size:  " << binary_data.size());
     const uint32_t crc_received =
         (*message_)[strings::msg_params][strings::crc32_check_sum].asUInt();
     LOG4CXX_TRACE(logger_, "CRC32 SUM Received: " << crc_received);
@@ -232,7 +231,7 @@ void PutFileRequest::Run() {
   }
 
   LOG4CXX_DEBUG(logger_,
-                "Writing " << bin_data_size << " bytes to " << full_path
+                "Writing " << binary_data.size() << " bytes to " << full_path
                            << " (current size is"
                            << file_system::FileSize(full_path) << ")");
 

--- a/src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_hmi_command_factory.cc
+++ b/src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_hmi_command_factory.cc
@@ -89,6 +89,7 @@ app_mngr::CommandSharedPtr VehicleInfoHmiCommandFactory::CreateCommand(
     message_type_str = "error response";
   }
 
+  UNUSED(message_type_str);
   LOG4CXX_DEBUG(logger_,
                 "HMICommandFactory::CreateCommand function_id: "
                     << function_id << ", message type: " << message_type_str);

--- a/src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_mobile_command_factory.cc
+++ b/src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_mobile_command_factory.cc
@@ -86,6 +86,7 @@ app_mngr::CommandSharedPtr VehicleInfoMobileCommandFactory::CreateCommand(
     message_type_str = "notification";
   }
 
+  UNUSED(message_type_str);
   LOG4CXX_DEBUG(logger_,
                 "HMICommandFactory::CreateCommand function_id: "
                     << function_id << ", message type: " << message_type_str);

--- a/src/components/application_manager/test/application_manager_impl_test.cc
+++ b/src/components/application_manager/test/application_manager_impl_test.cc
@@ -110,7 +110,9 @@ class ApplicationManagerImplTest : public ::testing::Test {
             application_manager::MockMessageHelper::message_helper_mock())
 
   {
+#ifdef ENABLE_LOG
     logger::create_log_message_loop_thread();
+#endif
     Mock::VerifyAndClearExpectations(mock_message_helper_);
   }
   ~ApplicationManagerImplTest() {

--- a/src/components/protocol_handler/src/protocol_handler_impl.cc
+++ b/src/components/protocol_handler/src/protocol_handler_impl.cc
@@ -290,6 +290,7 @@ void ProtocolHandlerImpl::SendStartSessionAck(
         static_cast<int64_t>(
             protocol_header_validator_.max_payload_size_by_service_type(
                 serviceTypeValue)));
+    UNUSED(mtu_written)
     LOG4CXX_DEBUG(logger_,
                   "MTU parameter was written to bson params: "
                       << mtu_written << "; Value: "
@@ -300,6 +301,7 @@ void ProtocolHandlerImpl::SendStartSessionAck(
       // Hash ID is only used in RPC case
       const bool hash_written = bson_object_put_int32(
           &params, strings::hash_id, static_cast<int32_t>(hash_id));
+      UNUSED(hash_written);
       LOG4CXX_DEBUG(logger_,
                     "Hash parameter was written to bson params: "
                         << hash_written << "; Value: "
@@ -317,6 +319,7 @@ void ProtocolHandlerImpl::SendStartSessionAck(
 
       const bool protocol_ver_written = bson_object_put_string(
           &params, strings::protocol_version, protocolVersionString);
+      UNUSED(protocol_ver_written);
       LOG4CXX_DEBUG(
           logger_,
           "Protocol version parameter was written to bson params: "

--- a/src/components/smart_objects/include/smart_objects/schema_item.h
+++ b/src/components/smart_objects/include/smart_objects/schema_item.h
@@ -39,6 +39,7 @@
 #include "smart_objects/errors.h"
 
 #include <memory>
+#include <vector>
 #include "utils/macro.h"
 #include "utils/semantic_version.h"
 

--- a/src/components/smart_objects/include/smart_objects/smart_object.h
+++ b/src/components/smart_objects/include/smart_objects/smart_object.h
@@ -35,6 +35,7 @@
 
 #include <set>
 #include <string>
+#include <sstream>
 #include <vector>
 #include <map>
 

--- a/src/components/transport_manager/src/iap2_emulation/iap2_transport_adapter.cc
+++ b/src/components/transport_manager/src/iap2_emulation/iap2_transport_adapter.cc
@@ -78,6 +78,7 @@ IAP2USBEmulationTransportAdapter::IAP2USBEmulationTransportAdapter(
   signal_handler_ = threads::CreateThread("iAP signal handler", delegate);
   signal_handler_->start();
   const auto result = mkfifo(out_signals_channel, mode);
+  UNUSED(result);
   LOG4CXX_DEBUG(logger_, "Out signals channel creation result: " << result);
 }
 
@@ -123,6 +124,7 @@ IAP2USBEmulationTransportAdapter::IAPSignalHandlerDelegate::
     IAPSignalHandlerDelegate(IAP2USBEmulationTransportAdapter& adapter)
     : adapter_(adapter), run_flag_(true), in_(0) {
   const auto result = mkfifo(in_signals_channel, mode);
+  UNUSED(result);
   LOG4CXX_DEBUG(logger_, "In signals channel creation result: " << result);
 }
 

--- a/src/components/transport_manager/src/tcp/platform_specific/linux/platform_specific_network_interface_listener.cc
+++ b/src/components/transport_manager/src/tcp/platform_specific/linux/platform_specific_network_interface_listener.cc
@@ -652,6 +652,7 @@ void PlatformSpecificNetworkInterfaceListener::DumpTable() const {
   for (auto it = status_table_.begin(); it != status_table_.end(); ++it) {
     const std::string ifname = it->first;
     const InterfaceStatus& status = it->second;
+    UNUSED(status);
 
     LOG4CXX_DEBUG(
         logger_,

